.github/workflows/check-pr-sync.yml

```
@@ -0,0 +1,32 @@
+  check-sync:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Fetch all branches
+        run: git fetch --all
+
+      - name: Check if branch is up-to-date with main
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:5:33: Double quote to prevent globbing and word splitting

(shellcheck)